### PR TITLE
[BritishCouncil] Add new extractor

### PR DESCRIPTION
**Pull request by jgilf**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

New extractor for britishcouncil.org videos as requested by #25368 

As the website uses Viddler as a provider, I've added a base Viddler IE to limit code repetition. 

All tests pass and code has been check with flake8.

### CODE REVIEW

**Comment by dstftw** [comment]
You must **delegate** to viddler extractor not inherit from it.

**Comment by jgilf** [comment]
Right, I understand. Sorry, didn't quite understand the convention.
I have updated my code to fix these issues.
Let me know if there is anything further I need to amend. Thanks!

**Comment by dstftw** [comment]
1. Read coding conventions.
2. Add support for [all kinds of embeds](https://support.viddler.com/hc/en-us/articles/206441496-Updating-Embed-Codes).

**Comment by jgilf** [comment]
Thanks, I've updated my regex to meet coding conventions. 

All three embed options in your link will now be caught by the generic extractor. Let me know any additional touch ups are required. Thanks!

File: youtube_dl/extractor/generic.py [hunk 1]
```diff
@@ -2580,6 +2595,12 @@ def _real_extract(self, url):
         if mobj is not None:
             return self.url_result(mobj.group('url'))
 
+        mobj = re.search(
+            r'<div[^>]+class=(["\'])viddler-auto-embed\1[^>]+data-video-id=([\'"])(?P<id>[^\'"]+)',
+            webpage)
+        if mobj is not None:
+            return ViddlerIE._build_url_result(mobj.group('id'))
+
```
Comment (dstftw) [inline youtube_dl/extractor/generic.py:2603]: Move embed extraction code to `ViddlerIE._extract_urls`.

File: youtube_dl/extractor/viddler.py [hunk 2]
```diff
@@ -74,6 +74,15 @@ class ViddlerIE(InfoExtractor):
...
     }]
 
+    @staticmethod
+    def _url_for_id(id):
+        return 'http://www.viddler.com/v/%s' % id
+
+    @classmethod
+    def _build_url_result(cls, id):
+        return cls.url_result(cls._url_for_id(id),
+                              ie=cls.ie_key())
```
Comment (dstftw) [inline youtube_dl/extractor/viddler.py:84]: Inline.

**Pull request closed without merging**